graphein/protein/visualisation.py

```diff
@@ -817,21 +824,61 @@ def asteroid_plot(
             node_x.append(x)
             node_y.append(y)
 
-        degrees = [
-            subgraph.degree(n) * node_size_multiplier for n in subgraph.nodes()
-        ]
+        def node_size_function(g: nx.Graph, feature: str):
+            if feature == 'degree':
+                return lambda k : g.degree(k)
+            elif feature == 'rsa':
+                return lambda k : g.nodes(data=True)[k]['rsa']
+            else:
+                raise NotImplementedError(f"Size by {size_nodes_by} not implemented.")
+        
+        node_size = node_size_function(subgraph, size_nodes_by)
+        node_sizes = [node_size_min + node_size(n) * node_size_multiplier for n in subgraph.nodes()]
 
         if colour_nodes_by == "shell":
             node_colours = []
             for n in subgraph.nodes():
                 for k, v in nodes.items():
                     if n in v:
                         node_colours.append(k)
+        elif colour_nodes_by == "hydrophobicity":
+
+            """
+            TODO Does a function like this already exist somewhere?
+            """
+            def hydrophobicity_of_residue(res: str, mapping: str = 'a'):
```

Comment:
Could you add the hydrophobicity values to `graphein.protein.resi_atoms` and add a reference to the source of the values. You could also add a node feature function to `graphein.protein.features.nodes` so these can be used elsewhere too :)